content/articles/how-to-build-an-nlp-based-emotion-detection-model-using-neattext-and-scikit-learn/index.md

```diff
@@ -0,0 +1,276 @@
+
+![Methods and attributes](/engineering-education/how-to-build-an-nlp-based-emotion-detection-model-using-neattext-and-scikit-learn/methods-attributes.png)
+
+We are interested in only two methods from the list: `remove_stopwords` and `remove_userhandles`.
+
+#### Removing user handles
+The dataset contains some Twitter handles of different users. This is noise to our dataset, and it's unnecessary. We use`remove_userhandles` to remove them from our dataset.
+
+```python
+df['Clean_Text'] = df['Text'].apply(nfx.remove_userhandles)
+```
+We use `apply()` to add `remove_userhandles`. We save the cleaned dataset into a new column named `Clean_Text`.
+
+#### Removing stopwords
+Stopwords is a list of all the commonly used words in any language. The words being common they have little classification power. These words may lead to model bias during training.
+Common stopwords are like articles of a given language.  In the English language they include: `the`, `is`, `and` and `are`.
+
+```python
+df['Clean_Text'] = df['Clean_Text'].apply(nfx.remove_stopwords)
+```
+In this section, we also use `apply()` to add `remove_stopwords`. We save the cleaned dataset into a new column named `Clean_Text`.
+
+To get the output of the clean dataset, run this command.
+
+```python
+df
+```
+The output of the dataset after removing user handles and stopwords is shown below.
+
+![Clean dataset](/engineering-education/how-to-build-an-nlp-based-emotion-detection-model-using-neattext-and-scikit-learn/clean-dataset.png)
+
+Now that we cleaned our dataset, let's load our machine learning packages.
+
+### Importing machine learning packages
+```python
+from sklearn.linear_model import LogisticRegression
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import accuracy_score
+```
+From the code above, we have imported the following.
+
+#### LogisticRegression
+`LogisticRegression` is an algorithm used for both classification and regression. This algorithm is imported from [Scikit-learn](https://scikit-learn.org/stable/). Will be used for emotion classification.
+
+#### CountVectorizer
+Machine learning models have a problem comprehending raw text. Machines cannot process the raw text data and it has to be converted into a matrix of numbers. Machines work well with numbers.
+
+CountVectorizer is used to convert the raw text into a matrix of numbers. This process depends on the frequency of each word in the entire text. During this process, `CountVectorizer` extracts important features from the text. They are then used as input for the model during training.
+
+#### train_test_split
+This method is important during the splitting of the dataset. It splits the dataset set into two sets, a train set, and a test set. This depends on the percentage specified by the user.
+
+#### accuracy_score
+This is important when calculating the accuracy score of our model during prediction.
```

Comment:
```suggestion
- `LogisticRegression` is an algorithm used for both classification and regression. This algorithm is imported from [Scikit-learn](https://scikit-learn.org/stable/). It will be used for emotion classification.

- Machine learning models have a problem comprehending raw text. Machines cannot process the raw text data and it has to be converted into a matrix of numbers. Machines work well with numbers. `CountVectorizer` is used to convert the raw text into a matrix of numbers. This process depends on the frequency of each word in the entire text. During this process, `CountVectorizer` extracts important features from the text. They are then used as input for the model during training.

- The `train_test_split` method is important during the splitting of the dataset. It splits the dataset set into two sets, a train set, and a test set. This depends on the percentage specified by the user.

- The `accuracy_score` is important when calculating the accuracy score of our model during prediction.
```